backend/core/utils/conf_lib.py

```diff
@@ -17,15 +17,20 @@ def create_prefix_defs(yaml_conf, prefixes):
         yaml_conf["prefixes"][prefix_str].yaml_set_anchor(prefix_str)
 
 
-def create_monitor_defs(yaml_conf):
+def create_monitor_defs(yaml_conf, monitors):
     """
     Create separate monitor definitions for the ARTEMIS conf file
     """
     yaml_conf["monitors"] = ruamel.yaml.comments.CommentedMap()
-    yaml_conf["monitors"]["riperis"] = [""]
-    yaml_conf["monitors"]["bgpstreamlive"] = ["routeviews", "ris"]
-    yaml_conf["monitors"]["betabmp"] = ["betabmp"]
-    # other monitors here
+    yaml_conf["monitors"]["riperis"] = monitors["riperis"]
```

Comment:
you should also add checks for what monitors are actually configured. What if there is not "riperis"? Add checks for key existence. Also it seems that this can be simplified in a loop in a much simpler, not hardcoded way:
```
for monitor in monitors:
    if monitor != "exabgp":
        yaml_conf["monitors"][monitor] = monitors[monitor]
    else:
        for dict_elem in monitors[monitor]:
            ...
```

Reply:
Ok, i will fix it

Reply:
@georgeepta: updates on this?

Reply:
@georgeepta updates on this?

Reply:
@vkotronis Sorry for the latency, last saturday i did the commit for the merging and i forgot to commit the conf_lib.py